simphony_mayavi/sources/slim_cuds_source.py

```diff
@@ -0,0 +1,543 @@
+    point_vectors_name = DEnum(values_name='_point_vectors_list',
+                               desc='vectors point data attribute to use')
+    cell_scalars_name = DEnum(values_name='_cell_scalars_list',
+                              desc='scalar cell data attribute to use')
+    cell_vectors_name = DEnum(values_name='_cell_vectors_list',
+                              desc='vectors cell data attribute to use')
+
+    #: Output information for the processing pipeline.
+    #: Overridden from the base class for more specialized setup.
+    output_info = PipelineInfo(
+        datasets=['image_data', 'poly_data', 'unstructured_grid'],
+        attribute_types=['any'],
+        attributes=['scalars', 'vectors'])
+
+    # The View to present in mayavi pane.
+    view = View(
+        Group(
+            Item(name='point_scalars_name'),
+            Item(name='point_vectors_name'),
+            Item(name='cell_scalars_name'),
+            Item(name='cell_vectors_name'),
+            Item(name='data')))
+
+    # Private traits #############################
+
+    # These private traits store the list of available data
+    # attributes for the above x_y_name enumeration.
+    _point_scalars_list = List(Str)
+    _point_vectors_list = List(Str)
+    _point_tensors_list = List(Str)
+    _cell_scalars_list = List(Str)
+    _cell_vectors_list = List(Str)
+    _cell_tensors_list = List(Str)
+
+    #: The shadow trait for the cuds property.
+    _cuds = Either(
+        Instance(ABCMesh),
+        Instance(H5Mesh),
+        Instance(ABCParticles),
+        Instance(ABCLattice))
+
+    #: The VTK backed CUDS container
+    _vtk_cuds = Either(
+        Instance(VTKMesh),
+        Instance(VTKParticles),
+        Instance(VTKLattice))
+
+    # The VTK dataset to manage. It is taken from the vtk_cuds.
+    # Name is unfortunately generic, but is for compatibility with
+    # CUDSSource. It is unsure if it needs to be public at all.
+    data = Property(Instance(tvtk.DataSet), depends_on="_vtk_cuds")
+
+    # This filter allows us to change the attributes of the data
+    # object and will ensure that the pipeline is properly taken care
+    # of.  Directly setting the array in the VTK object will not do
+    # this.
+    _assign_attribute = Instance(tvtk.AssignAttribute, args=(),
+                                 allow_none=False)
+
+    # The ID of the observer for the data.
+    _observer_id = Int(-1)
+
+    def __init__(self, cuds=None, point_scalars=None, point_vectors=None,
+                 cell_scalars=None, cell_vectors=None, **traits):
+        """ Constructor
+
+        Parameters
+        ----------
+        cuds : ABCParticles, ABCLattice, ABCMesh or H5Mesh
+            The CUDS dataset to be wrapped as VTK data source
+
+        point_scalars : str
+            CUBA name of the data to be selected as point scalars.
+            Default is the first available point scalars.
```

Comment:
Yes it is. It will pick the first one available (first being in alphabetical order), and if not, the empty selection, which means it will display nothing for that data.